### PR TITLE
Expand development docs and rationale

### PR DESCRIPTION
**Pull request by bgilbert**
Rendered version [here](https://bgilbert.github.io/ignition/).

### CODE REVIEW

**Comment by bgilbert** [comment]
I think this one can skip release notes.

**Comment by bgilbert** [comment]
cc @dustymabe @prestist

**Comment by travier** [comment]
Looks great. Thanks!

**Comment by travier** [comment]
Merging as this is a doc only change.

File: docs/development.md [hunk 1]
```diff
@@ -18,6 +18,80 @@ sudo apt-get install libblkid-dev
...
+1. OS enables networking if required
+1. `fetch` stage
+1. OS examines fetched config and e.g. copies root filesystem contents to RAM
+1. `disks` stage
+1. OS mounts root filesystem
+1. `mount` stage
+1. OS does any preprocessing of configured filesystems, including copying root filesystem contents back
+1. `files` stage
+1. OS does any postprocessing of provisioned system
+1. `umount` stage
```
Comment (dustymabe) [inline docs/development.md:85]: Would be nice if the description of each stage is a sub-bullet. Right now it looks a bit awkward:

Screenshot 2023-08-11 at 10-33-48 Development



Also would be nice if we could add a description for the umount stage like all the others even though it's a bit superfluous.

**Review by dustymabe — COMMENTED** [review]
LGTM I only had one suggestion.

File: docs/development.md (hunk 1 above)
Comment (bgilbert) [reply docs/development.md:85]: Those aren't descriptions of the stages; they're descriptions of what happens between the stages.  So it's rendering correctly.

Comment (dustymabe) [reply docs/development.md:85]: ahh - I see. I should have read more closely.

**Pull request merged**